File: py/URI_RDFS.py
"""
===============================================================================
Creator : Ceri Binding, University of South Wales | Prifysgol de Cymru 
Contact : ceri.binding@southwales.ac.uk
Project : Any
Classes : URI_RDFS
Summary : Case-sensitive entity & property URI constants (for RDF apps) 
Example : 
    import URI_RDFS
    print(URI_RDFS.CLASS) # "http://www.w3.org/2000/01/rdf-schema#Class"
License : https://mit-license.org
History :
23/02/2023 CFB Extracted from old script, updated to use F-strings
===============================================================================
"""
# Namespace
NS = "http://www.w3.org/2000/01/rdf-schema#"

# Entities
RESOURCE = f"{NS}Resource"
LITERAL = f"{NS}Literal"
CLASS = f"{NS}Class"
DATATYPE = f"{NS}Datatype"
CONTAINER = f"{NS}Container"
CONTAINERMEMBERSHIPPROPERTY = f"{NS}ContainerMembershipProperty"

# Properties
SUBCLASSOF = f"{NS}subClassOf"
SUBPROPERTYOF = f"{NS}subPropertyOf"
DOMAIN = f"{NS}domain"
RANGE = f"{NS}range"
LABEL = f"{NS}label"
COMMENT = f"{NS}comment"
MEMBER = f"{NS}member"
SEEALSO = f"{NS}seeAlso"
ISDEFINEDBY = f"{NS}isDefinedBy"

File: js/URI_RDFS.js
/*
===============================================================================
Creator : Ceri Binding, University of South Wales | Prifysgol de Cymru 
Contact : ceri.binding@southwales.ac.uk
Project : Any
Classes : URI_RDFS 
Summary : Case-sensitive entity & property URI constants (for RDF apps) 
Example : 
    <script type="module">
        import {URI_RDFS} from "./URI_RDFS.js"
        console.log(URI_RDFS.LITERAL) | console.log(URI_RDFS["LITERAL"])
        # both return "http://www.w3.org/2000/01/rdf-schema#Literal"
    </script>
License : https://mit-license.org
History :
10/02/2023 CFB Extracted from legacy script and converted to ES6 module
===============================================================================
*/
const NS = "http://www.w3.org/2000/01/rdf-schema#" 
export const URI_RDFS = Object.freeze({
    // entities
    CLASS: `${NS}Class`,
    CONTAINER: `${NS}Container`,
    CONTAINERMEMBERSHIPPROPERTY: `${NS}ContainerMembershipProperty`,
    DATATYPE: `${NS}Datatype`,
    LITERAL: `${NS}Literal`,
    RESOURCE: `${NS}Resource`,
    // properties
    COMMENT: `${NS}comment`,
    DOMAIN: `${NS}domain`,
    ISDEFINEDBY: `${NS}isDefinedBy`,
    LABEL: `${NS}label`,
    MEMBER: `${NS}member`,
    RANGE: `${NS}range`,
    SEEALSO: `${NS}seeAlso`,
    SUBCLASSOF: `${NS}subClassOf`,
    SUBPROPERTYOF: `${NS}subPropertyOf`
})